tests/MongoDB.Driver.Tests/Linq/Translators/PredicateTranslatorTests.cs

```diff
@@ -147,6 +144,48 @@ public void Any_with_a_nested_Any()
                 "{G: {$elemMatch: {D: \"Don't\", \"S\": {$elemMatch: {E: null, D: \"Delilah\" }}}}}");
         }
 
+        [Fact]
+        public void Any_with_a_not_and_a_predicate_with_not_contains()
+        {
+            var x = new[] { 1, 2 };
+
+            Assert(
+                c => !c.M.Any(a => !x.Contains(a)),
+                "{ M : { '$not' : { '$elemMatch' : { '$nin' : [1, 2] } } } }");
```

Comment:
I'm concerned that we've transformed the expression so much and that we may have inadvertently altered the semantics of the query. Why not translate more directly to:

    { $nor : [ { M : { $elemMatch : { $not : { $in : [1, 2] } } } } ] }

Note: $nor at top level is because this is the only to do a $not in MQL (when not in an operator expression).

Reply:
So, there are two points to mention here:
1. The current implementation of `$nor` operation requires having the first element in query which is started on "$". In your query: 
>{ $nor : [ { M : { $elemMatch : { $not : { $in : [1, 2] } } } } ] }
the first element in a query (which should be negated) is `M`. 
So, to have `nor` in result query,  as I see this part requires bigger changes which potentially can affect some other code parts.
So, `$not` operation here is just because of the previous implementation. 
I guess It can be changed somehow If it's necessary. 

2. About which part is better: `{ '$nin' : [1, 2] }` or `{ $not : { $in : [1, 2] } }`.
Do you think there is a significant difference in these parts? The implemented way looked better to me, but I can change this part as well.

Reply:
@rstam, is this point resolved?